src/ansi.rs

```diff
@@ -742,19 +767,46 @@ impl<'a, H, W> vte::Perform for Performer<'a, H, W>
         _ignore: bool,
         byte: u8
     ) {
+        macro_rules! unhandled {
+            () => {{
+                err_println!("[unhandled] esc_dispatch params={:?}, ints={:?}, byte={:?} ({:02x})",
+                             params, intermediates, byte as char, byte);
+                return;
+            }}
+        }
+
+        let i1: u8 = match intermediates.first() {
```

Comment:
I think it would make sense to move this into `deg_charset!` since it's not used for anything else. It could then be written like
```rust
macro_rules! deg_charset {
    ($charset:path) => {{
        let index = match intermediates.first() {
            Some(b'(') => CharsetIndex::G0,
            Some(b')') => CharsetIndex::G1,
            Some(b'*') => CharsetIndex::G2,
            Some(b'+') => CharsetIndex::G3,
            _ => unhandled!(),
        };
        self.handler.designate_charset(index, $charset);
    }}
}
```